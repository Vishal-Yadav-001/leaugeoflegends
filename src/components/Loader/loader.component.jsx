import { LinearProgress } from "@mui/material";
import React, { Fragment } from "react";
import './loader.styles.scss'
/**
 * Displays the progress / scale 
 * @param {string} variant 
 * @param {string} value  - The value to display and to show it on scale
 * @param {string} sx - custom styles for mui  
 * @param {string} size - size of progrss bar 
 * @returns Linear Progress UI
 */
const Loader = ({ variant, value, sx, size, thickness, color }) => {
  return (
    <Fragment>
      <LinearProgress
        size={size}
        thickness={thickness}
        variant={variant}
        value={value}
        sx={sx}
        color={color}
        className="progress"
        data-testid="loader"
      />
    </Fragment>
  );
};

export default Loader;
