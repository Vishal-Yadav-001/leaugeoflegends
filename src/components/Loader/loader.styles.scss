@import '../../styles/variables.styles.scss';
.MuiLinearProgress-root.MuiLinearProgress-determinate{
    height:10px;
    border-radius: 50px;
    width:100%;
    background-color: $primary;
}
.MuiLinearProgress-bar.MuiLinearProgress-barColorPrimary{
    height:10px;
    border-radius: 50px;
    width:100%;
    background-color: $accent;
}