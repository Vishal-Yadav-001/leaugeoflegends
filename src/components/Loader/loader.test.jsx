import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Loader from "./loader.component";

describe("Render Loader", () => {
  test("<Loader/>", () => {
    const loader = render(<Loader variant="indeterminate" />, {
      wrapper: BrowserRouter,
    });
    expect(screen.getByTestId("loader")).toBeInTheDocument();
  });
  test("<Loader/>Snapshot", () => {
    const loader = renderer
      .create(
        <BrowserRouter>
          <Loader variant="indeterminate" />
        </BrowserRouter>
      )
      .toJSON();
    expect(loader).toMatchSnapshot();
  });
});
