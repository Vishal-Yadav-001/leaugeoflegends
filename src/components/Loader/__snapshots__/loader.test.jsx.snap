// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Loader <Loader/>Snapshot 1`] = `
<span
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate progress css-eglki6-MuiLinearProgress-root"
  data-testid="loader"
  role="progressbar"
>
  <span
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
    style={Object {}}
  />
  <span
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
    style={Object {}}
  />
</span>
`;
